siliconcompiler/core.py

```diff
@@ -574,12 +593,28 @@ def create_cmdline(self,
                         self.error(f"Invalid value '{item}'' for switch {switchstr}. "
                                    "Too many arguments, please wrap multiline strings in quotes.")
                         continue
-                    elif len(items) == 3:
-                        step, index, val = items
-                    elif len(items) == 2:
-                        step, val = items
+                    if sctype == 'bool':
+                        if len(items) == 3:
+                            step, index, val = items
+                        elif len(items) == 2:
+                            step, val = items
+                            if val != 'true' and val != 'false':
+                                index = val
+                                val = True
+                        elif len(items) == 1:
+                            val, = items
+                            if val != 'true' and val != 'false':
+                                step = val
```

Comment:
Is there a check if this step exists at this point?

Reply:
not until `run()`

Reply:
Ok if it's checked at that point that is also fine.